### PR TITLE
feat(cli): Add Splunk alert channel

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Splunk Alert Channel

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: cli/cmd/integration.go [hunk 1]
```diff
@@ -444,6 +447,34 @@ func reflectIntegrationData(raw api.RawIntegration) [][]string {
...
+		if err != nil {
+			cli.Log.Debugw("unable to decode integration data",
+				"integration_type", raw.Type,
+				"raw_data", raw.Data,
+				"error", err,
+			)
+			break
+		}
+		out := [][]string{
+			[]string{"Channel", iData.Channel},
```
Comment (afiune) [inline cli/cmd/integration.go:463]:
```suggestion
			[]string{"CHANNEL", iData.Channel},
```

File: api/integration_alert_channels_splunk.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+// UpdateSplunkAlertChannel updates a single splunk alert channel integration
+func (svc *IntegrationsService) UpdateSplunkAlertChannel(data SplunkAlertChannel) (
+	response SplunkAlertChannelResponse,
+	err error,
+) {
+	err = svc.update(data.IntgGuid, data, &response)
+	return
+}
+
+// ListSplunkAlertChannel lists the SPLUNK external integrationS available on the Lacework Server
```
Comment (afiune) [inline api/integration_alert_channels_splunk.go:85]:
```suggestion
// ListSplunkAlertChannel lists the SPLUNK external integrations available on the Lacework Server
```

File: cli/cmd/integration_splunk.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		{
+			Name:   "ssl",
+			Prompt: &survey.Confirm{Message: "Enable SSL?"},
+		},
+	}
+
+	answers := struct {
+		Name     string
+		Channel  string
+		HecToken string
```
Comment (afiune) [inline cli/cmd/integration_splunk.go:72]: Do you need a survey annotation here? 🤔 could you test that it loads all the fields, please.

File: api/integration_alert_channels_splunk.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	commonIntegrationData
+	Data SplunkChannelData `json:"DATA"`
+}
+
+type SplunkChannelData struct {
+	Channel   string          `json:"CHANNEL,omitempty" mapstructure:"CHANNEL"`
+	HecToken  string          `json:"HEC_TOKEN" mapstructure:"HEC_TOKEN"`
+	Host      string          `json:"HOST" mapstructure:"HOST"`
+	Port      int             `json:"PORT" mapstructure:"PORT"`
+	Ssl       bool            `json:"SSL,omitempty" mapstructure:"SSL"`
```
Comment (afiune) [inline api/integration_alert_channels_splunk.go:107]: I don't see any default on the API Schema:
```
            "SSL": {
              "enumNames": [
                "Enabled",
                "Disabled"
              ],
              "title": "SSL",
              "type": "boolean"
            }
```

Can we really omit this field?

**Review by afiune — APPROVED** [review]
nitpicks but overall +1

tenor-227119993

File: cli/cmd/integration_splunk.go (hunk 3 above)
Comment (dmurray-lacework) [reply cli/cmd/integration_splunk.go:72]: Seems to work ok, but I'm going to add in the survey annotation for hectoken regardless

File: api/integration_alert_channels_splunk.go (hunk 4 above)
Comment (dmurray-lacework) [reply api/integration_alert_channels_splunk.go:107]: Fixed, I've removed omit empty

**Pull request merged**